src/Microsoft.HttpRepl.Tests/FileSystem/RealFileSystemTests.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.HttpRepl.FileSystem;
+using Xunit;
+
+namespace Microsoft.HttpRepl.Tests.FileSystem
+{
+    public class RealFileSystemTests
+    {
+        [Theory]
+        [InlineData(".json")]
+        [InlineData(".xml")]
+        [InlineData(".tmp")]
+        public void GetTempFileName_WithExtension_ReturnsFileNameWithExtension(string extension)
```

Comment:
Should this have a WithoutExtension test too?

Reply:
never mind, I see you removed that overload.  Does WithExtension need to be in the test name?

Reply:
I changed it to WithValidInput, since I also created WithInvalidInput tests. That seemed better than just removing it.